src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/AvoidZeroLengthArrayAllocationsTests.cs

```diff
@@ -708,5 +708,36 @@ class C
 #pragma warning restore RS0030 // Do not used banned APIs
                 fixedSource);
         }
+
+        [Fact]
+        [WorkItem(4665, "https://github.com/dotnet/roslyn-analyzers/issues/4665")]
+        public async Task NoDiagnosticInExpressionTree_CSharp()
+        {
+            const string source = @"
+using System;
+using System.Linq.Expressions;
+
+class C
+{
+    Expression<Func<int[]>> f = () => new int[0];
+}
+";
+            await VerifyCS.VerifyCodeFixAsync(source, source);
+        }
+
+        [Fact]
+        [WorkItem(4665, "https://github.com/dotnet/roslyn-analyzers/issues/4665")]
+        public async Task NoDiagnosticInExpressionTree_VisualBasic()
```

Comment:
🔴 This test has passed before fixing the issue. It shouldn't have passed. Need to figure out why this happened.